apisix/plugins/mocking.lua

```diff
@@ -0,0 +1,211 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+local core = require("apisix.core")
+local ngx = ngx
+local xml2lua = require("xml2lua")
+local string = string
+local type = type
+local table = table
+local math = math
+local pairs = pairs
+
+local schema = {
+    type = "object",
+    properties = {
+        -- specify response delay time,default 0ms
+        delay = { type = "integer", default = 0 },
+        -- specify response status,default 200
+        response_status = { type = "integer", default = 200, minimum = 1 },
+        -- specify response content type, support application/xml, text/plain
+        -- and application/json, default application/json
+        content_type = { type = "string", default = "application/json" },
```

Comment:
Better to use `enum` to define the support type.

Reply:
> Better to use `enum` to define the support type.

@starsz because the content_type support two parts type and charset such as: "application/json" and "application/json;charset=utf8",so it's hard to define a enum~